app/src/main/java/com/example/habitsmasher/PublicPrivateButtons.java

```diff
@@ -0,0 +1,131 @@
+package com.example.habitsmasher;
+
+import static android.content.ContentValues.TAG;
+
+import android.nfc.Tag;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+
+import com.google.android.material.tabs.TabLayout;
+
+public class PublicPrivateButtons {
+    /**
+     * This class handles the logic behind the public and private buttons, which are used to
+     * set a habit to either public or private.
+     *
+     * @author Cameron Matthew
+     * @version 1.0
+     */
+
+    private Button _publicButton;
+    private Button _privateButton;
+    private boolean _publicSelected;
+    private boolean _privateSelected;
+
+    /**
+     * This constructor is the default constructor, and is used in {@link com.example.habitsmasher.ui.dashboard.AddHabitDialog}.
+     * @param view The view that has the buttons.
+     */
+    public PublicPrivateButtons(View view){
```

Comment:
Constructor for adding habits

Reply:
I'm not 100% sure about the title of PublicPrivateButtons. I think that maybe a name like StatusButtons or something along those lines makes more sense? This is just my personal preference though so not a huge thing by any means